File: Linear Data Structure/LinkedList.java
//Main linkedlist class
public class LinkedList {
   
    Node head; 
   
    static class Node {
   
        int data;
        Node next;
   

        Node(int d){
            data = d;
            next = null;
    }


}

//insert function implemented 
public static LinkedList insert(LinkedList list, int data){
    
    Node new_node = new Node(data);
    new_node.next = null;
   
    
    if (list.head == null) {
        list.head = new_node;
    }
    else {
        Node last = list.head;
        while (last.next != null) {
            last = last.next;
        }
        last.next = new_node;
    }
   
    return list;
}
   

// for printing the linkedlist
public static void printList(LinkedList list){
    Node node = list.head;
    
    System.out.print("LinkedList: ");

    while (node != null) {
            
        System.out.print(node.data +" ");

        node = node.next;
    }

}



    
//main funtion
public static void main(String[] args){
    //linkedlist object created
    LinkedList list = new LinkedList();
   
   
    // Inserting values in the linkedlist
        
    list = insert(list, 14);
    list = insert(list, 52);
    list = insert(list, 6);
    list = insert(list, 21);
    list = insert(list, 2);
    list = insert(list, 53);
    list = insert(list, 7);
    list = insert(list, 8);
   
    printList(list);

    }
}


/*
Output:

LinkedList: 14 52 6 21 2 53 7 8 

*/

File: Linear Data Structure/LinkedList.py
"""
A simple implementation of Linear datastructure Linkedlist.

"""

class Node:

    def __init__(self, value) -> None:
        self.val = value
        self.next = None


class LinkedList:

    def __init__(self) -> None:
        
        self.head = None
        self.tail = None

    def printll(self):

        temp = self.head

        while temp != None:
            print(temp.val, end = " ")

            temp = temp.next

        print()

    def push(self, val, flag):
        temp = Node(val)

        if flag == 0:
            temp.next = self.head
            self.head = temp
        else:
            self.tail.next = temp
            self.tail = temp
             

    def insert(self,pre ,post, node):

        pre.next = node
        node.next = post


        

if __name__ == "__main__":

    node1 = Node(1)
    node2 = Node(2)
    node3 = Node(3)

    node4 = Node(4)
    node5 = Node(5)

    node1.next = node2
    node2.next = node3
    node3.next = node4
    node4.next = node5

    node6 = Node(10)

    node7 = Node(15)

    ll = LinkedList()

    ll.head = node1
    ll.tail = node5

    ll.printll()

    ll.push(0,0)
    ll.push(6,1)

    ll.insert(node2, node3, node6)
    ll.insert(node2, node6, node7)


    ll.printll()


"""
# Output

    1 2 3 4 5 
    0 1 2 15 10 3 4 5 6 
    
"""
